File: vokimi_api/Src/dtos/responses/test_taking/general/GeneralTestTakingData.cs
using vokimi_api.Src.db_related.db_entities.published_tests.general_test_related;
using vokimi_api.Src.dtos.responses.test_taking.general;
using vokimi_api.Src.enums;

namespace vokimi_api.Src.dtos.responses.test_taking
{
    public record class GeneralTestTakingData(
        string TestTemplate,
        string AccentColor,
        string ArrowIcons,
        TestTakingConclusionData? ConclusionData,
        GeneralTestTakingQuestionData[] Questions
    ) : ITestTakingDataResponse
    {
        public static GeneralTestTakingData FromGeneralTest(TestGeneralTemplate test) => new(
            test.Template.GetId(),
            test.StylesSheet.AccentColor,
            test.StylesSheet.ArrowsType.GetId(),
            test.Conclusion is null ? null : TestTakingConclusionData.FromConclusion(test.Conclusion),
            test.Questions.Select(GeneralTestTakingQuestionData.FromQuestion).ToArray()
        );
    }
}

File: vokimi_frontend/src/ts/page_classes/test_taking_page/general_test/GeneralTestTakingData.ts
import type { TestStylesArrowType } from "../../../enums/TestStylesArrowType";
import { TestTemplate } from "../../../enums/TestTemplate";
import type { ITestTakingData, TestTakingConclusionData } from "../ITestTakingData";
import type { GeneralTestTakingQuestionData } from "./GeneralTestTakingQuestionData";

export class GeneralTestTakingData implements ITestTakingData {
    readonly testTemplate: TestTemplate = TestTemplate.General;
    readonly accentColor: string;
    readonly arrowIcons: TestStylesArrowType;
    readonly conclusion: TestTakingConclusionData;
    readonly questions: GeneralTestTakingQuestionData[];

    constructor(
        accentColor: string,
        arrowIcons: TestStylesArrowType,
        conclusion: TestTakingConclusionData,
        questions: GeneralTestTakingQuestionData[]
    ) {
        this.accentColor = accentColor;
        this.arrowIcons = arrowIcons;
        this.conclusion = conclusion;
        this.questions = questions;
    }
}
